lib/rust/frp/src/stream.rs

```diff
@@ -659,24 +659,24 @@ where Def: InputBehaviors
     }
 }
 
-// FIXME code quality below:
 impl<Def> HasOutputTypeLabel for Node<Def>
 where Def: HasOutputStatic + InputBehaviors
 {
-    fn output_type_label(&self) -> String {
-        let label = type_name::<Def>().to_string();
-        let label = label.split(|c| c == '<').collect::<Vec<_>>()[0];
-        let mut label = label.split(|c| c == ':').collect::<Vec<_>>();
-        label.reverse();
-        let mut label = label[0];
-        let sfx = "Data";
-        if label.ends_with(sfx) {
-            label = &label[0..label.len() - sfx.len()];
-        }
-        label.into()
+    fn output_type_label(&self) -> Label {
+        type_name_to_output_label(type_name::<Def>())
     }
 }
 
+// The label transformation logic is a separate non-generic function, so that it can be compiled
+// only once for all node types. This has a noticeable impact on compilation time.
```

Comment:
add here a link to this PR if it mentions "noticeable impact on compile time" - it's worth proving statements :)